src/routes/movies.py

```
@@ -1,13 +1,266 @@
+            joinedload(MovieModel.genres),
+            joinedload(MovieModel.actors),
+            joinedload(MovieModel.languages),
+        )
+        .where(MovieModel.id == movie_id)
+    )
+    res = await db.execute(stmt)
+    movie = res.unique().scalar_one_or_none()
+    if movie is None:
+        raise HTTPException(
+            status_code=404, detail="Movie with the given ID was not found."
+        )
+    return _serialize_full(movie)
+
+
+@router.delete("/{movie_id}/", status_code=status.HTTP_204_NO_CONTENT)
+async def delete_movie(movie_id: int, db: AsyncSession = Depends(get_db)):
+    res = await db.execute(select(MovieModel).where(MovieModel.id == movie_id))
+    movie = res.scalar_one_or_none()
+    if movie is None:
+        raise HTTPException(
+            status_code=404, detail="Movie with the given ID was not found."
+        )
+    await db.delete(movie)
+    await db.commit()
+    return
+
 
+@router.patch("/{movie_id}/")
+async def update_movie(
+    movie_id: int,
+    payload: MovieUpdateSchema,
+    db: AsyncSession = Depends(get_db),
+):
+    res = await db.execute(select(MovieModel).where(MovieModel.id == movie_id))
+    movie = res.scalar_one_or_none()
+    if movie is None:
+        raise HTTPException(
+            status_code=404, detail="Movie with the given ID was not found."
+        )
 
-router = APIRouter()
+    try:
+        if payload.name is not None:
+            movie.name = payload.name
+        if payload.date is not None:
+            movie.date = payload.date
+        if payload.score is not None:
+            movie.score = payload.score
+        if payload.overview is not None:
+            movie.overview = payload.overview
+        if payload.status is not None:
+            movie.status = payload.status
+        if payload.budget is not None:
+            movie.budget = payload.budget
+        if payload.revenue is not None:
+            movie.revenue = payload.revenue
+    except Exception:
```

The `try/except Exception:` in the PATCH handler is too broad. Catching all exceptions may hide unexpected errors and makes debugging harder. Prefer validating inputs via Pydantic (and map ValidationError to 400) or catching specific exceptions (e.g. `ValueError`, `TypeError`) if you expect them. This will still let unexpected errors surface to logs/tests and avoids masking bugs.